### DISCUSSION TITLE
Variables in devcontainer.json?

### DISCUSSION THREAD

**Discussion by jalanh11640** [devcontainers/community#142] [Q&A]
Is there anyway to specify a variable in devcontainer.json?  I'm thinking something like this..

"variables": {
     "HOST": "myHostName"
},
{ // devcontainer.json starts here
...
    "runArgs": [
        "-h", "${variables:HOST}"
    ],
    "postCreateCommand": "vscodePostCreate.sh ${variables:HOST}"
}

Thanks,
Jeff

**Comment by samruddhikhandale** [comment]
Hi 👋 

Are you familiar with `containerEnv` and `remoteEnv`?
Look at https://containers.dev/implementors/json_schema/ for more details/

**Reply by jalanh11640** [reply]
Thanks for the reply.  I wasn't familiar with those.  However, looking at the documentation, neither of those look like they'll support what I'm trying to do.  I have several places in my devcontainer.json where I've used the same value.  I was hoping to find a way to specify the value once, and use it multiple places within the devcontainer.json file.

Jeff
